aixplain/model_interfaces/interfaces/function_models.py

```diff
@@ -179,3 +194,207 @@ def predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Di
             TextToImageGenerationOutput(**text_to_image_generation_dict)
             text_to_image_generation_output["predictions"][i] = text_to_image_generation_dict
         return text_to_image_generation_output
+
+    
+class TextGenerationModel(AixplainModel):
+    def run_model(self, api_input: Dict[str, List[TextGenerationInput]], headers: Dict[str, str] = None) -> Union[Dict[str, List[TextGenerationOutput]], Dict[str, List[List[int]]]]:
+        pass
+    
+    def _route(self, request):
+        if "function" in request.keys():
+            function_type = request["function"].upper()
+            if function_type == "PREDICT":
+                return self._predict
+            elif function_type == "TOKENIZE":
+                return self._run_tokenize
+            else:
+                raise ValueError("Invalid function.")
+        else:
+            return self._predict
+
+    def predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
+        function = self._route(request)
+        return function(request, headers)
+
+    def _predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
+        instances = request['instances']
+        text_generation_input_list = []
+        # Convert JSON serializables into TextGenerationInputs
+        for instance in instances:
+            text_generation_input = TextGenerationInput(**instance)
+            text_generation_input_list.append(text_generation_input)
+        text_generation_output = self.run_model({"instances": text_generation_input_list})
+
+        # Convert JSON serializables into TextGenerationOutputs
+        for i in range(len(text_generation_output["predictions"])):
+            text_generation_dict = text_generation_output["predictions"][i].dict()
+            TextGenerationOutput(**text_generation_dict)
+            text_generation_output["predictions"][i] = text_generation_dict
+        return text_generation_output
+    
+    def _run_tokenize(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
+        instances = request['instances']
+        tokenization_input_list = []
+        # Convert JSON serializables into TextListInputs
+        for instance in instances:
+            tokenization_input = TextListInput(**instance)
```

Comment:
You can remove this instance validation by adding a [validate_call](https://docs.pydantic.dev/latest/api/validate_call/#pydantic.validate_call_decorator.validate_call) decorator on tokenize method, which would then validate it and raise error if it doesn't comply

Reply:
I would actually use that everywhere as it makes the code much cleaner, but we would need to remove ambiguous types like Dict from the predict endpoint